packages/data-store/src/identifier/did-store.ts

```diff
@@ -138,7 +149,7 @@ export class DIDStore extends AbstractDIDStore {
       const service = new Service()
       service.id = argsService.id
       service.type = argsService.type
-      service.serviceEndpoint = argsService.serviceEndpoint
+      service.serviceEndpoint = JSON.stringify(argsService.serviceEndpoint)
```

Comment:
There should be a type check here, because if `argsService.serviceEndpoint` is already a string, it will get serialized as `"serviceEndpoint"` instead of `serviceEndpoint` (quotes added).